drivers: nxp_enet_experimental: Build error when using PTP

**Describe the bug**
NXP enet experimental does not build with ptp on

**To Reproduce**
Steps to reproduce the behavior:
1. cd to zephyr base
2. west build -b mimxrt1050_evk -p always samples/net/gptp/ -DEXTRA_DTC_OVERLAY_FILE=nxp,enet-experimental.overlay
3. observe build error

```
[1/338] Preparing syscall dependency handling

[3/338] Generating include/generated/version.h
-- Zephyr version: 3.5.99 (/home/declan/zephyrproject/zephyr), build: zephyr-v3.5.0-4574-g4158d9523ea9
[188/338] Building C object zephyr/drivers/ptp_c.../drivers__ptp_clock.dir/ptp_clock_nxp_enet.c.obj
FAILED: zephyr/drivers/ptp_clock/CMakeFiles/drivers__ptp_clock.dir/ptp_clock_nxp_enet.c.obj
ccache /home/declan/zephyr-sdk-0.16.3/arm-zephyr-eabi/bin/arm-zephyr-eabi-gcc -DBOARD_FLASH_SIZE="CONFIG_FLASH_SIZE*1024" -DCPU_MIMXRT1052DVL6B -DENET_ENHANCEDBUFFERDESCRIPTOR_MODE -DKERNEL -DK_HEAP_MEM_POOL_SIZE=0 -DNDEBUG -DPICOLIBC_DOUBLE_PRINTF_SCANF -DXIP_BOOT_HEADER_DCD_ENABLE=1 -DXIP_BOOT_HEADER_ENABLE=1 -DXIP_EXTERNAL_FLASH -D_POSIX_C_SOURCE=200809 -D__LINUX_ERRNO_EXTENSIONS__ -D__PROGRAM_START -D__ZEPHYR_SUPERVISOR__ -D__ZEPHYR__=1 -I/home/declan/zephyrproject/zephyr/include -I/home/declan/zephyrproject/zephyr/build/zephyr/include/generated -I/home/declan/zephyrproject/zephyr/soc/arm/nxp_imx/rt -I/home/declan/zephyrproject/zephyr/lib/posix/getopt/. -I/home/declan/zephyrproject/zephyr/subsys/net/l2 -I/home/declan/zephyrproject/zephyr/subsys/net/lib/dns/. -I/home/declan/zephyrproject/modules/hal/cmsis/CMSIS/Core/Include -I/home/declan/zephyrproject/zephyr/modules/cmsis/. -I/home/declan/zephyrproject/modules/hal/nxp/mcux/mcux-sdk/devices/MIMXRT1052 -I/home/declan/zephyrproject/modules/hal/nxp/mcux/mcux-sdk/devices/MIMXRT1052/drivers -I/home/declan/zephyrproject/modules/hal/nxp/mcux/mcux-sdk/devices/MIMXRT1052/xip -I/home/declan/zephyrproject/modules/hal/nxp/mcux/mcux-sdk/drivers/common -I/home/declan/zephyrproject/modules/hal/nxp/mcux/mcux-sdk/drivers/trng -I/home/declan/zephyrproject/modules/hal/nxp/mcux/mcux-sdk/drivers/enet -I/home/declan/zephyrproject/modules/hal/nxp/mcux/mcux-sdk/drivers/lpuart -I/home/declan/zephyrproject/modules/hal/nxp/mcux/mcux-sdk/drivers/igpio -I/home/declan/zephyrproject/modules/hal/nxp/mcux/mcux-sdk/drivers/cache/armv7-m7 -I/home/declan/zephyrproject/zephyr/modules/hal_nxp/. -isystem /home/declan/zephyrproject/zephyr/lib/libc/common/include -Wshadow -fno-strict-aliasing -Os -imacros /home/declan/zephyrproject/zephyr/build/zephyr/include/generated/autoconf.h -fno-common -g -gdwarf-4 -fdiagnostics-color=always -mcpu=cortex-m7 -mthumb -mabi=aapcs -mfp16-format=ieee -mtp=soft --sysroot=/home/declan/zephyr-sdk-0.16.3/arm-zephyr-eabi/arm-zephyr-eabi -imacros /home/declan/zephyrproject/zephyr/include/zephyr/toolchain/zephyr_stdint.h -Wall -Wformat -Wformat-security -Wno-format-zero-length -Wno-pointer-sign -Wpointer-arith -Wexpansion-to-defined -Wno-unused-but-set-variable -Werror=implicit-int -fno-pic -fno-pie -fno-asynchronous-unwind-tables -ftls-model=local-exec -fno-reorder-functions --param=min-pagesize=0 -fno-defer-pop -fmacro-prefix-map=/home/declan/zephyrproject/zephyr/samples/net/gptp=CMAKE_SOURCE_DIR -fmacro-prefix-map=/home/declan/zephyrproject/zephyr=ZEPHYR_BASE -fmacro-prefix-map=/home/declan/zephyrproject=WEST_TOPDIR -ffunction-sections -fdata-sections --specs=picolibc.specs -D_POSIX_THREADS -std=c99 -MD -MT zephyr/drivers/ptp_clock/CMakeFiles/drivers__ptp_clock.dir/ptp_clock_nxp_enet.c.obj -MF zephyr/drivers/ptp_clock/CMakeFiles/drivers__ptp_clock.dir/ptp_clock_nxp_enet.c.obj.d -o zephyr/drivers/ptp_clock/CMakeFiles/drivers__ptp_clock.dir/ptp_clock_nxp_enet.c.obj -c /home/declan/zephyrproject/zephyr/drivers/ptp_clock/ptp_clock_nxp_enet.c
/home/declan/zephyrproject/zephyr/drivers/ptp_clock/ptp_clock_nxp_enet.c: In function 'nxp_enet_ptp_clock_callback':
/home/declan/zephyrproject/zephyr/drivers/ptp_clock/ptp_clock_nxp_enet.c:159:41: error: 'data' redeclared as different kind of symbol
  159 |         struct ptp_clock_nxp_enet_data *data = dev->data;
      |                                         ^~~~
/home/declan/zephyrproject/zephyr/drivers/ptp_clock/ptp_clock_nxp_enet.c:156:31: note: previous definition of 'data' with type 'void *'
  156 |                         void *data)
      |                         ~~~~~~^~~~
/home/declan/zephyrproject/zephyr/drivers/ptp_clock/ptp_clock_nxp_enet.c:186:43: error: incompatible types when assigning to type 'struct k_mutex' from type 'struct k_mutex *'
  186 |                 *(struct k_mutex *)data = &data->ptp_mutex;
      |                                           ^
[195/338] Building C object zephyr/drivers/ether...keFiles/drivers__ethernet.dir/eth_nxp_enet.c.obj
/home/declan/zephyrproject/zephyr/drivers/ethernet/eth_nxp_enet.c: In function 'eth_nxp_enet_rx':
/home/declan/zephyrproject/zephyr/drivers/ethernet/eth_nxp_enet.c:423:17: warning: implicit declaration of function 'ptp_clock_nxp_enet_get' [-Wimplicit-function-declaration]
  423 |                 ptp_clock_nxp_enet_get(config->ptp_clock, &ptp_time);
      |                 ^~~~~~~~~~~~~~~~~~~~~~
[197/338] Building C object zephyr/subsys/net/ip/CMakeFiles/subsys__net__ip.dir/tcp.c.obj
ninja: build stopped: subcommand failed.
FATAL ERROR: command exited with status 1: /usr/bin/cmake --build /home/declan/zephyrproject/zephyr/build
```

--- a/drivers/ethernet/eth_nxp_enet.c
+++ b/drivers/ethernet/eth_nxp_enet.c
@@ -39,6 +39,7 @@ LOG_MODULE_REGISTER(LOG_MODULE_NAME);
 #include <zephyr/dt-bindings/ethernet/nxp_enet.h>
 #include <zephyr/net/phy.h>
 #include <zephyr/net/mii.h>
+#include <zephyr/drivers/ptp_clock.h>
 #if defined(CONFIG_NET_DSA)
 #include <zephyr/net/dsa.h>
 #endif
@@ -351,7 +352,7 @@ static int eth_nxp_enet_rx(const struct device *dev)
 	uint16_t vlan_tag = NET_VLAN_TAG_UNSPEC;
 	uint32_t frame_length = 0U;
 	struct net_if *iface;
-	struct net_pkt *pkt;
+	struct net_pkt *pkt = NULL;
 	status_t status;
 	uint32_t ts;
 
@@ -420,7 +421,7 @@ static int eth_nxp_enet_rx(const struct device *dev)
 	if (eth_get_ptp_data(get_iface(data, vlan_tag), pkt)) {
 		struct net_ptp_time ptp_time;
 
-		ptp_clock_nxp_enet_get(config->ptp_clock, &ptp_time);
+		ptp_clock_get(config->ptp_clock, &ptp_time);
 
 		/* If latest timestamp reloads after getting from Rx BD,
 		 * then second - 1 to make sure the actual Rx timestamp is accurate
@@ -453,7 +454,9 @@ flush:
 					0, RING_ID, NULL);
 	__ASSERT_NO_MSG(status == kStatus_Success);
 error:
-	net_pkt_unref(pkt);
+	if (pkt) {
+		net_pkt_unref(pkt);
+	}
 	eth_stats_update_errors_rx(get_iface(data, vlan_tag));
 	return -EIO;
 }

--- a/drivers/ptp_clock/ptp_clock_nxp_enet.c
+++ b/drivers/ptp_clock/ptp_clock_nxp_enet.c
@@ -153,7 +153,7 @@ static int ptp_clock_nxp_enet_rate_adjust(const struct device *dev,
 
 void nxp_enet_ptp_clock_callback(const struct device *dev,
 			enum nxp_enet_callback_reason event,
-			void *data)
+			void *cb_data)
 {
 	const struct ptp_clock_nxp_enet_config *config = dev->config;
 	struct ptp_clock_nxp_enet_data *data = dev->data;
@@ -181,9 +181,9 @@ void nxp_enet_ptp_clock_callback(const struct device *dev,
 				      &ptp_config);
 	}
 
-	if (data != NULL) {
+	if (cb_data != NULL) {
 		/* Share the mutex with mac driver */
-		*(struct k_mutex *)data = &data->ptp_mutex;
+		*(uintptr_t *)cb_data = (uintptr_t)&data->ptp_mutex;
 	}
 }
 

--- a/samples/net/gptp/sample.yaml
+++ b/samples/net/gptp/sample.yaml
@@ -21,3 +21,10 @@ tests:
     depends_on: netif
     integration_platforms:
       - frdm_k64f
+  sample.net.gpt.nxp_enet_experimental:
+    extra_args: EXTRA_DTC_OVERLAY_FILE="nxp,enet-experimental.overlay"
+    platform_allow:
+      - mimxrt1050_evk
+      - mimxrt1060_evk
+      - mimxrt1064_evk
+      - mimxrt1024_evk

